src/ripple/server/WSSession.h

```diff
@@ -140,6 +140,10 @@ struct WSSession
     void
     close() = 0;
 
+    virtual
+    void
+    close(const std::uint16_t& code, const std::string& reason) = 0;
```

Comment:
Instead of passing the `code` and `reason`, could be pass `boost::beast::websocket::close_reason` instead?

(also, Nit: We prefer to take integer types like `std::uint16_t` by value instead of `const&`)